### PR TITLE
R.chain transucer, closes #1033

### LINKED ISSUE #1033: support transducers in R.chain
```javascript
R.into([], R.map(function(x) { return [x, x]; }), [1, 2, 3]);
// => [[1, 1], [2, 2], [3, 3]]

R.into([], R.chain(function(x) { return [x, x]; }), [1, 2, 3]);
//                 acc = xf.step(acc, list[idx]);
//                          ^
// TypeError: undefined is not a function
```

### PR DESCRIPTION
**Pull request by buzzdecafe**
h/t to @kevinbeaty for his help pushing this over the finish line

### CODE REVIEW

File: src/internal/_chain.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var _curry2 = require('./_curry2');
+var map = require('../map');
+var unnest = require('../unnest');
+
+
+module.exports = _curry2(function chain(f, list) {
```
Comment (davidchambers) [inline src/internal/_chain.js]: This function should be named `_chain`.

File: src/internal/_flatCat.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  return function _xcat(xf) {
+    var rxf = preservingReduced(xf);
+    return {
+      "@@transducer/init": _xfBase.init,
+      "@@transducer/result": function(result) {
+        return rxf["@@transducer/result"](result);
+      },
+      "@@transducer/step": function(result, input) {
+        return (!isArrayLike(input)) ? _reduce(rxf, result, [input]) : _reduce(rxf, result, input);
```
Comment (davidchambers) [inline src/internal/_flatCat.js]: The parens around `!isArrayLike(input)` are unnecessary.

**Comment by davidchambers** [comment]
Could you fix the lint errors?

**Comment by buzzdecafe** [comment]
hmmm. passes locally

**Comment by buzzdecafe** [comment]
((:bell:))

**Comment by davidchambers** [comment]
:deciduous_tree:

**Pull request merged**